mail_layout_preview/wizard/email_template_preview.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2020 Simone Orsi - Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+
+
+class MailTemplatePreview(models.TransientModel):
+    _inherit = "mail.template.preview"
+
+    _url_pattern = "/email-preview/{model}/{templ_id}/{rec_id}/"
+
+    layout_preview_url = fields.Char(
+        string="Full layout preview", compute="_compute_layout_preview_url"
+    )
+
+    @api.depends("resource_ref", "model_id", "mail_template_id")
+    def _compute_layout_preview_url(self):
+        for rec in self:
+            if rec.mail_template_id:
```

Comment:
```suggestion
            if rec.mail_template_id and rec.resource_ref:
```
There are some template that we don't have a record to render preview